pkg/cloudprovider/plugins/azure/azure_resource_graph_nwIntfTable.go

```diff
@@ -77,19 +77,24 @@ func getNetworkInterfaceTable(resourceGraphAPIClient azureResourceGraphWrapper,
 	if err != nil {
 		return nil, 0, err
 	}
-	if data == nil {
+	if len(data) == 0 {
 		return []*networkInterfaceTable{}, 0, nil
 	}
 
 	var networkInterfaces []*networkInterfaceTable
-	networkInterfaceRows := data.([]interface{})
-	for _, networkInterfaceRow := range networkInterfaceRows {
-		var networkInterface networkInterfaceTable
-		err = mapstructure.Decode(networkInterfaceRow, &networkInterface)
-		if err != nil {
-			return nil, 0, err
+	for _, page := range data {
```

Comment:
I wonder if you can avoid making changes in the caller, and let the logic of combining page data in invokeResourceGraphQuery api itself

Reply:
Logic in caller is specific to the type(virtulamachine/NetworkInterface), so decoding each page should stay in the caller itself.